cranelift/codegen/src/isa/x86/mod.rs

```diff
@@ -189,6 +189,15 @@ impl TargetIsa for Isa {
     fn as_any(&self) -> &dyn Any {
         self as &dyn Any
     }
+
+    fn is_compatible_with(&self, other: &dyn TargetIsa) -> bool {
+        match other.as_any().downcast_ref::<Self>() {
```

Comment:
I'm not sure if the question makes sense, but is there a chance that since the x64 new backend uses a different TargetIsa, this downcast may fail when comparing on a native x86 host?